### DISCUSSION TITLE
why font render not use the canvas original font solution?

### DISCUSSION THREAD

**Discussion by bobby-ai** [ONLYOFFICE/DocumentServer#1837] [Q&A] [answered]
Taking the spreadsheet as an example, Why does font rendering need to be manually rendered using the freetype library, which is compiled into a wasm package and executed by front-end loading。

I notice that Javascript and css provide font-face technology to load custom fonts,  and  provide 
 the CanvasRenderingContext2D.font property of the Canvas 2D API to specify the current text style and draw text，which is simple and easy to maintain，and the  canvas's bottom layer consider font shaping and freetype technology, which is better than the current source solution

**Comment by ShockwaveNN** [comment, marked as answer]
Hi, first of all - I'm not the developer of our font engine, but I think I can provide some reasoning
I think the main reason for our rather complicated technology stack - then we've started around 10 years ago with DocumentServer project there was no proper implementation of font rendering via JavaScript
Another reason - by our contracts, we need to support rather old systems like Windows XP and IE, so they may have trouble with modern solutions
I think we got some plans to move to something modern, but we need to redo most of our font engine, which, right now, renders almost any file almost perfectly (which cannot do any other web editors as far as I know)

**Reply by bobby-ai** [reply]
Thanks, I think the analysis is reasonable, and expect modern plans and solution
